ListUtil.xs

```diff
@@ -952,11 +952,13 @@ PPCODE:
                     stack[reti++] = newSVsv(PL_stack_base[i + 1]);
         }
 
-        if (spill)
+        if (spill) {
             /* the POP_MULTICALL will trigger the SAVEFREESV above;
              * keep it alive  it on the temps stack instead */
             SvREFCNT_inc_simple_void_NN(spill);
-            sv_2mortal((SV*)spill);
+        }
+
+        sv_2mortal((SV*)spill);
```

Comment:
This looks wrong. I believe the original logic intention was that `sv_2mortal()` also be conditional on whether `spill` is true. As it happens, that line was executed unconditionally, and apparently that it doesn't crash all the time is simply because `sv_2mortal()` is apparently OK to be called on NULL. But nevertheless, the call ought to be inside the conditional braces.